ts/components/conversation/ImageGrid.tsx

```diff
@@ -1,263 +1,142 @@
-            />
-          </div>
-        </div>
-      </div>
-    );
-  }
-
-  const moreMessagesOverlay = attachments.length > 5;
-  const moreMessagesOverlayText = moreMessagesOverlay ? `+${attachments.length - 5}` : undefined;
+  const columnImageSide = THUMBNAIL_SIDE / 2 - 5;
 
+  // we know only support having 3 attachments displayed at most.
   return (
-    <div className="module-image-grid">
-      <div className="module-image-grid__column">
-        <div className="module-image-grid__row">
-          <Image
-            alt={getAlt(attachments[0])}
-            attachment={attachments[0]}
-            playIconOverlay={isVideoAttachment(attachments[0])}
-            height={149}
-            width={149}
-            url={isMessageVisible ? getThumbnailUrl(attachments[0]) : undefined}
-            onClick={onClickAttachment}
-            onError={onError}
-            attachmentIndex={0}
-          />
-          <Image
-            alt={getAlt(attachments[1])}
-            playIconOverlay={isVideoAttachment(attachments[1])}
-            height={149}
-            width={149}
-            attachment={attachments[1]}
-            url={isMessageVisible ? getThumbnailUrl(attachments[1]) : undefined}
-            onClick={onClickAttachment}
-            onError={onError}
-            attachmentIndex={1}
-          />
-        </div>
-        <div className="module-image-grid__row">
-          <Image
-            alt={getAlt(attachments[2])}
-            bottomOverlay={withBottomOverlay}
-            playIconOverlay={isVideoAttachment(attachments[2])}
-            height={99}
-            width={99}
-            attachment={attachments[2]}
-            url={isMessageVisible ? getThumbnailUrl(attachments[2]) : undefined}
-            onClick={onClickAttachment}
-            onError={onError}
-            attachmentIndex={2}
-          />
-          <Image
-            alt={getAlt(attachments[3])}
-            bottomOverlay={withBottomOverlay}
-            playIconOverlay={isVideoAttachment(attachments[3])}
-            height={99}
-            width={98}
-            attachment={attachments[3]}
-            url={isMessageVisible ? getThumbnailUrl(attachments[3]) : undefined}
-            onClick={onClickAttachment}
-            onError={onError}
-            attachmentIndex={3}
-          />
-          <Image
-            alt={getAlt(attachments[4])}
-            bottomOverlay={withBottomOverlay}
-            playIconOverlay={isVideoAttachment(attachments[4])}
-            height={99}
-            width={99}
-            darkOverlay={moreMessagesOverlay}
-            overlayText={moreMessagesOverlayText}
-            attachment={attachments[4]}
-            url={isMessageVisible ? getThumbnailUrl(attachments[4]) : undefined}
-            onClick={onClickAttachment}
-            onError={onError}
-            attachmentIndex={4}
-          />
-        </div>
-      </div>
-    </div>
+    <StyledImageGrid>
+      <Image
```

Comment:
Same here with using map instead of 3 images components. Probably you could combine them and then pass through the attachments array

Reply:
:+1: